src/assert.js

```diff
@@ -48,8 +48,10 @@ QUnit.assert = Assert.prototype = {
 		};
 	},
 
-	// Exports test.push() to the user API
-	push: function( /* result, actual, expected, message, negative */ ) {
+	// Exports test.pushResult() to the user API
+	pushResult: function( resultInfo ) {
```

Comment:
The implementation looks great, but we need to keep a deprecated `.push` method until 2.0 is released.

the deprecated `.push()` can be a simple abstraction to call `.pushResult` with the given arguments.